app/ts/App.tsx

```diff
@@ -149,38 +149,39 @@ export function App() {
 	const repBalance = useOptionalSignal<EthereumQuantity>(undefined)
 	const daiBalance = useOptionalSignal<EthereumQuantity>(undefined)
 
+	const pathSignal = useSignal<string>('')
+
 	const tabs = [
 		{ title: 'Trading', path: 'trading', component: <DeployContract maybeAccountAddress = { maybeAccountAddress } areContractsDeployed = { areContractsDeployed }/> },
 		{ title: 'Market Creation', path: 'market-creation', component: <CreateYesNoMarket maybeAccountAddress = { maybeAccountAddress } universe = { universe } reputationTokenAddress = { reputationTokenAddress }/> },
 		{ title: 'Reporting', path: 'reporting', component: <Reporting maybeAccountAddress = { maybeAccountAddress } universe = { universe } reputationTokenAddress = { reputationTokenAddress }/> },
 		{ title: 'Claim Funds', path: 'claim-funds', component: <ClaimFunds maybeAccountAddress = { maybeAccountAddress }/> },
-		{ title: 'Migration', path: 'migration', component: <Migration maybeAccountAddress = { maybeAccountAddress } reputationTokenAddress = { reputationTokenAddress } universe = { universe } universeForkingInformation = { universeForkingInformation } /> }
+		{ title: 'Migration', path: 'migration', component: <Migration maybeAccountAddress = { maybeAccountAddress } reputationTokenAddress = { reputationTokenAddress } universe = { universe } universeForkingInformation = { universeForkingInformation } pathSignal = { pathSignal }/> }
 	] as const
 
-	useEffect(() => {
-		const hash = window.location.hash.replace('#/', '')
-		const [path, params] = hash.split('?')
-		const tabIndex = tabs.findIndex(tab => tab.path === path)
+	useSignalEffect(() => {
+		window.history.pushState({}, '', pathSignal.value)
+		const hash = pathSignal.value.replace('#/', '')
+		const [pathPart, params] = hash.split('?')
```

Comment:
You could save a line if you wanted.
```suggestion
		const [pathPart, params] = pathSignal.value.replace('#/', '').split('?')
```